server-common/src/main/java/org/apache/kafka/purgatory/DeferredEventPurgatory.java

```diff
@@ -26,10 +26,9 @@
 
 /**
  * The purgatory which holds events that have been started, but not yet completed.
- * We wait for the high water mark of the metadata log to advance before completing
- * them.
+ * We wait for the high watermark of the log to advance before completing them.
  */
-class ControllerPurgatory {
+public class DeferredEventPurgatory {
```

Comment:
Maybe a more accurate name could be `OffsetBasedCompletion`

Reply:
Naming things is hard :). `Completion` sounds a bit weird here. I can't really think of a better name...